aws-android-sdk-iot/src/test/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManagerTest.java

```diff
@@ -3021,6 +3021,38 @@ public void testEnableMetricsCollection() {
         assertFalse(testClient.isMetricsEnabled());
     }
 
+    @Test
+    public void testMqttSessionPresent() throws MqttException, InterruptedException {
+        // Test if sessionPresent flag from AWSIotMqttManager is correctly passed from the MqttClient
+        MockMqttClient mockClient = new MockMqttClient();
+        final boolean actualSessionPresent;
+        actualSessionPresent = mockClient.testToken.getSessionPresent();
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        final AWSIotMqttManager testSessionPresentFlag = new AWSIotMqttManager("test-sessionPresent-flag",
+                Region.getRegion(Regions.US_EAST_1), TEST_ENDPOINT_PREFIX);
+        testSessionPresentFlag.setMqttClient(mockClient);
+
+        KeyStore testKeystore = AWSIotKeystoreHelper.getIotKeystore(CERT_ID, KEYSTORE_PATH,
+                KEYSTORE_NAME, KEYSTORE_PASSWORD);
+
+        testSessionPresentFlag.connect(testKeystore, new AWSIotMqttClientStatusCallback() {
+            @Override
+            public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
+                if (status == AWSIotMqttClientStatus.Connecting || status == AWSIotMqttClientStatus.Connected) {
+                    System.out.println("Client persistent-client-id connection status: " + status);
+                    System.out.println("getSessionPresent: " + testSessionPresentFlag.getSessionPresent());
+                    countDownLatch.countDown();
+                }
+            }
+        });
+
+        if (countDownLatch.await(2, TimeUnit.SECONDS)) {
+            // Compare sessionPresent flag from AWSIotMqttManager with the actual one
+            assertEquals(testSessionPresentFlag.getSessionPresent(), actualSessionPresent);
+        }
```

Comment:
```suggestion
        if (countDownLatch.await(2, TimeUnit.SECONDS)) {
            // Compare sessionPresent flag from AWSIotMqttManager with the actual one
            assertEquals(testSessionPresentFlag.getSessionPresent(), actualSessionPresent);
        } else {
            fail("CountDownLatch timed out.");
        }
```

If the latch times out, then the `await` will return `false` and never assert anything, causing the test to pass erroneously.